Mathlib/AlgebraicGeometry/PrimeSpectrum/Basic.lean

```diff
@@ -844,6 +849,34 @@ lemma exists_idempotent_basicOpen_eq_of_isClopen {s : Set (PrimeSpectrum R)}
 @[deprecated (since := "2024-11-11")]
 alias exists_idempotent_basicOpen_eq_of_is_clopen := exists_idempotent_basicOpen_eq_of_isClopen
 
+section localization_quotient
+variable [CommSemiring R] [CommSemiring S]
+
+open Localization Polynomial in
+lemma comap_C_eq_comap_localization_union_comap_quotient
+    {R : Type*} [CommRing R] (s : Set (PrimeSpectrum R[X])) (c : R) :
+    comap C '' s =
+      comap (algebraMap R (Away c)) '' (comap C ''
+        (comap (mapRingHom (algebraMap R (Away c))) ⁻¹' s)) ∪
+      comap (Ideal.Quotient.mk (.span {c})) '' (comap C ''
+        (comap (mapRingHom (Ideal.Quotient.mk _)) ⁻¹' s)) := by
```

Comment:
I think it's better to replace this by a lemma about Set only that could go into a much earlier file, namely that if you have `f` and `g` that are jointly surjective, and commutative squares `h ∘ f = f₁ ∘ f₂` and `h ∘ g = g₁ ∘ g₂`, then `h '' s = f₁ '' (f₂ '' (f ⁻¹' s)) ∪ g₁ '' (g₂ '' (g ⁻¹' s))`; the three conditions can be proved locally when applying the lemma.
Of course, if you are going to use this lemma multiple times, it would be worth keeping it too.

Reply:
You're right, that's quite neater